openhtf/plugs/__init__.py

```diff
@@ -151,24 +150,53 @@ class BasePlug(object):
   enable_remote = False
   # Allow explicitly disabling remote access to specific attributes.
   disable_remote_attrs = set()
-  # Default logger to be used only in __init__ of subclasses..
+  # Default logger to be used only in __init__ of subclasses.
   # This is overwritten both on the class and the instance so don't store
   # a copy of it anywhere.
   logger = _LOG
 
   @classproperty
   def placeholder(cls):
-    """Returns a PlugPlaceholder for calling class."""
+    """Returns a PlugPlaceholder for the calling class."""
     return PlugPlaceholder(cls)
 
+  @classmethod
+  def get_descriptor(cls):
```

Comment:
not sure I want this exposed to plug implementors? We limit the attributes we set here to ones that make sense for implementors to set/override (except for `placeholder` which has to be here). But getting the name, mro, and descriptor should be up to the `PlugManager` since we don't want to have hard-to-debug failures of the frontend because someone decided to implement `get_descriptor()` or `get_name()` as normal plug functions

Reply:
Done